app/scripts/views/form.js

```diff
@@ -199,6 +199,10 @@ define(function (require, exports, module) {
 
     _submitForm: notifyDelayedRequest(showButtonProgressIndicator(function () {
       var self = this;
+      // sets all password fields to type=password
+      self.$el.find('.password').each(function (i, el) {
```

Comment:
> Adding it there will need an event handler for submit as well, which is related to the form, and not the password field. I think it would be better to keep it here.

mmm.. I'm trying to be pragmatic, and can understand your argument. At the same time, my feeling is that separating view show/hide password logic in this way is going to cause confusion in the future.

It seems like the proper thing to do is for the form to send a "hey, we are about to submit!" message out to entities that care before [`beforeSubmit`](https://github.com/mozilla/fxa-content-server/pull/3832/files#diff-1115514375ece0df300f7c60b8f56426R207). People that care (like the password mixin) could take some action, like convert all `.password` fields to `[type=password]`. That would keep password view logic out of form.js, and it keeps the core submit logic out of the password-mixin. I'd say use `beforeSubmit`, but that has a different purpose, which is to check whether the form should be submit.

The message I refer to could be a literal message using "this.emit", or it could be a function call that can be overridden.